operator/pkg/helmreconciler/prune.go

```diff
@@ -78,6 +91,105 @@ func (h *HelmReconciler) Prune(manifests name.ManifestMap, all bool) error {
 	})
 }
 
+// PruneControlPlaneByRevisionWithController is called to remove specific control plane revision in specific namespace
+// during reconciliation process of controller.
+// It returns the install status and any error encountered.
+func (h *HelmReconciler) PruneControlPlaneByRevisionWithController(ns, revision string) (*v1alpha1.InstallStatus, error) {
```

Comment:
i have a suggestion here... you have PruneControlPlaneByRevision and DeleteControlPlaneByManifests. the first function doesn't seem to do what the name implies because you're passing in the objects to delete. 
how about having two functions called GetResourcesByRevision and GetResourcesFromManifest, select one of these depending on the filename flag, and at the end always call what is now (incorrectly named) pruneUnlistedResources? i think that will make the flow a bit easier to follow.

Reply:
updated the naming